fundingmanager.go

```diff
@@ -2276,33 +2264,7 @@ func (f *fundingManager) handleFundingLocked(fmsg *fundingLockedMsg) {
 		f.barrierMtx.Unlock()
 	}()
 
-	// Finally, we'll find the peer that sent us this message so we can
-	// provide it with the fully initialized channel state.
-	peer, err := f.cfg.FindPeer(fmsg.peerAddress.IdentityKey)
-	if err != nil {
-		fndgLog.Errorf("Unable to find peer: %v", err)
-		channel.Stop()
-		return
-	}
-	newChanDone := make(chan struct{})
-	newChanMsg := &newChannelMsg{
-		channel: channel,
-		done:    newChanDone,
-	}
-
-	select {
-	case peer.newChannels <- newChanMsg:
-	case <-f.quit:
-		return
-	}
-
-	// We pause here to wait for the peer to recognize the new channel
-	// before we close the channel barrier corresponding to the channel.
-	select {
-	case <-f.quit:
-		return
-	case <-newChanDone: // Fallthrough if we're not quitting.
-	}
+	fmsg.peer.AddNewChannel(channel, f.quit)
```

Comment:
Does it make sense to let this return an error that we can log here?

Reply:
Fixed.